### PR TITLE
Unify mongodb test stat477

### PR DESCRIPTION
**Pull request by cj-oci**
Integrate eosiod_run_mongodb_test.sh into eosiod_run_test.py. Currently test method getTransByBlockId() has been disabled as it needs to handle multiple transactions. That is an extra validation step and will be done in a followup PR. Create a single test script for both master and Noon branch for maintenance ease.

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -67,7 +67,8 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/restart-scenarios-test.py ${CMAKE_CUR
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/testUtils.py ${CMAKE_CURRENT_BINARY_DIR}/testUtils.py COPYONLY)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/eosiod_run_test.py ${CMAKE_CURRENT_BINARY_DIR}/eosiod_run_test.py COPYONLY)
 
-add_test(NAME eosiod_run_test COMMAND tests/eosiod_run_test.py --exitEarly --dumpErrorDetail WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+add_test(NAME eosiod_run_test COMMAND tests/eosiod_run_test.py --noon --exit-early --dump-error-detail WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+add_test(NAME eosiod_run_test COMMAND tests/eosiod_run_test.py --noon --mongodb --exit-early --dump-error-detail WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
```
Comment (heifner) [inline tests/CMakeLists.txt]: ctest does not like 2 tests with the same name.

File: tests/eosiod_run_test.py [hunk 2]
```diff
@@ -105,8 +113,6 @@ def cmdError(name, code=0, exitNow=False):
     exchangeAccount.ownerPrivateKey=PRV_KEY2
     exchangeAccount.ownerPublicKey=PUB_KEY2
     
-    time.sleep(5)
-
     Print("Stand up walletd")
     if walletMgr.launch() is False:
         cmdError("eos-walletd")
```
Comment (heifner) [inline tests/eosiod_run_test.py]: eosio-walletd

File: tests/eosiod_run_test.py [hunk 3]
```diff
@@ -323,25 +329,48 @@ def cmdError(name, code=0, exitNow=False):
...
+    transaction=None
+    if not enableMongo:
+        transaction=node.getTransaction(transId)
+    else:
+        if amINoon:
+            transaction=node.getActionFromDb(transId)
+        else:
+            transaction=node.getMessageFromDb(transId)
     if transaction is None:
         cmdError("eosc get transaction trans_id")
```
Comment (heifner) [inline tests/eosiod_run_test.py]: eosioc

File: tests/eosiod_run_test.py [hunk 4]
```diff
@@ -323,25 +329,48 @@ def cmdError(name, code=0, exitNow=False):
...
+            amountVal=transaction["data"]["amount"]
+
     if typeVal!= "transfer" or amountVal != 975311:
-    #if transaction.tType != "transfer" or transaction.amount != 975311:
         errorExit("FAILURE - get transaction trans_id failed: %s" % (transId), raw=True)
 
     Print("Get transactions for account %s" % (testeraAccount.name))
     actualTransactions=node.getTransactionsArrByAccount(testeraAccount.name)
     if actualTransactions is None:
         cmdError("eosc get transactions testera")
```
Comment (heifner) [inline tests/eosiod_run_test.py]: Please search and change all `eosc` to `eosioc`, `eosd` to `eosiod`.  And `eos-walletd` to  `eosio-walletd` or `eosiowd` which will be the new name I believe.

File: tests/testUtils.py [hunk 5]
```diff
@@ -20,11 +20,12 @@
...
     FNull = open(os.devnull, 'w')
-    EosClientPath="programs/eosioc/eosioc"
-    EosWalletPath="programs/eosio-walletd/eosio-walletd"
-    EosServerName="eosiod"
-    EosServerPath="programs/eosiod/%s" % (EosServerName)
+    EosClientPath="programs/eosc/eosc"
+    EosWalletPath="programs/eos-walletd/eos-walletd"
+    EosServerName="eosd"
+    EosServerPath="programs/eosd/%s" % (EosServerName)
     EosLauncherPath="programs/launcher/launcher"
```
Comment (heifner) [inline tests/testUtils.py]: Looks like this diff went the wrong way. The removed lines are correct.

File: tests/eosiod_run_test.py (hunk 4 above)
Comment (cj-oci) [reply tests/eosiod_run_test.py]: Valid comment on executable name change. Made the change.

File: tests/testUtils.py (hunk 5 above)
Comment (cj-oci) [reply tests/testUtils.py]: This script is identical for master and noon branch. This section defines the master specific names. Method iAmNoon() sets the Noon specific names. I did this cause it was becoming a pain-point to sync the script between the two branches. Once noon becomes master we will get rid of this bridge code.

**Comment by heifner** [comment]
6/8 Test #6: distributed-transactions-test ..........***Failed    0.51 sec
terminate called after throwing an instance of 'fc::eof_exception'
unrecognised option '--eosd'

**Pull request merged**